src/commands/createFunction/FunctionListStep.ts

```diff
@@ -49,8 +49,8 @@ export class FunctionListStep extends AzureWizardPromptStep<IFunctionWizardConte
                 if (options.templateId) {
                     const actualId: string = t.id.toLowerCase();
                     const expectedId: string = options.templateId.toLowerCase();
-                    // Check for a match with or without the language in the id
-                    return actualId === expectedId || actualId === `${expectedId}-${language.toLowerCase()}`;
+                    // check for only whole word matches and escape the "\b" since it's in a string, not a RegExp expression
+                    return new RegExp(`\\b${expectedId}\\b`, 'gmi').test(actualId);
```

Comment:
The RegExp looks good to me, but please keep `actualId === expectedId` in the check. You're passing in "HttpTrigger", but I also want this to continue working if people pass in the exact id (aka "HttpTrigger-JavaScript" or "Azure.Function.CSharp.HttpTrigger.3.x") instead of the shortened/generic id